### PR TITLE
Add hcloud_server_network module

### PR DESCRIPTION
**Pull request by LKaemmerling**
##### SUMMARY

Follow Up from #59366 and Part 4/4 of #59133

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

hcloud_server_network
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/modules/cloud/hcloud/hcloud_server_network.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = """
+---
+module: hcloud_server_network
+
+short_description: Manage the relationship between Networks and servers
```
Comment (resmo) [inline lib/ansible/modules/cloud/hcloud/hcloud_server_network.py]: s/Networks/networks

File: lib/ansible/modules/cloud/hcloud/hcloud_server_network.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+
+version_added: "2.9"
+
+description:
+    - Create, update and delete cloud subnetworks on the Hetzner Cloud.
```
Comment (resmo) [inline lib/ansible/modules/cloud/hcloud/hcloud_server_network.py]: s/subnetworks/networks

based on code review, the module does not update anything, right? Remove the update?

File: lib/ansible/modules/cloud/hcloud/hcloud_server_network.py [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+
+author:
+    - Lukas Kaemmerling (@lkaemmerling)
+
+options:
+    network:
+        description:
+            - The name of the Hetzner Cloud Networks.
```
Comment (resmo) [inline lib/ansible/modules/cloud/hcloud/hcloud_server_network.py:34]: s/Networks/network

File: lib/ansible/modules/cloud/hcloud/hcloud_server_network.py [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+            - The name of the Hetzner Cloud server.
+        type: str
+        required: true
+    ip:
+        description:
+            - The IP the server should have.
+        type: str
+    alias_ips:
+        description:
+            - Alias IPs the Server has.
```
Comment (resmo) [inline lib/ansible/modules/cloud/hcloud/hcloud_server_network.py]: s/Server/server

File: lib/ansible/modules/cloud/hcloud/hcloud_server_network.py [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+"""
+
+RETURN = """
+hcloud_server_network:
+    description: The relationship between a server and a network
+    returned: always
+    type: complex
+    contains:
+        network:
+            description: Name of the Network
```
Comment (resmo) [inline lib/ansible/modules/cloud/hcloud/hcloud_server_network.py:102]: s/Network/network

File: lib/ansible/modules/cloud/hcloud/hcloud_server_network.py [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+            type: str
+            returned: always
+            sample: my-network
+        server:
+            description: Name of the Server
```
Comment (resmo) [inline lib/ansible/modules/cloud/hcloud/hcloud_server_network.py]: s/Server/server

**Review by resmo — CHANGES_REQUESTED** [review]
minor doc issues, otherwise no complains :)

File: lib/ansible/modules/cloud/hcloud/hcloud_server_network.py (hunk 5 above)
Comment (LKaemmerling) [reply lib/ansible/modules/cloud/hcloud/hcloud_server_network.py:102]: The product is called `Network` so we would leave it as it is.

File: lib/ansible/modules/cloud/hcloud/hcloud_server_network.py (hunk 3 above)
Comment (LKaemmerling) [reply lib/ansible/modules/cloud/hcloud/hcloud_server_network.py:34]: The product is called `Network` so we would leave it as it is.

File: lib/ansible/modules/cloud/hcloud/hcloud_server_network.py (hunk 1 above)
Comment (LKaemmerling) [reply lib/ansible/modules/cloud/hcloud/hcloud_server_network.py]: The product is called `Network` so we would leave it as it is.

**Review by resmo — APPROVED** [review]
shipit

**Comment by cschmitt-hcloud** [comment]
shipit

**Pull request merged**